### ISSUE TITLE
Support Snowflake SSO authentication

### ISSUE THREAD

**Issue by gioruss** [trinodb/trino#21373]
Hi,
I'm trying to use Trino Snowflake connector. I noticed that, at the moment, the only supported authentication optio is the internal Snowflake authenticator (username and password required).
However, in many use cases it would be necessary to take advantage of SSO authentication, provided by the Snowflake JDBC driver.

Thanks

**Comment by BrandonNoad** [comment]
Yes, we would like to have the ability to use Snowflake OAuth for authentication, which requires setting the `authenticator` and `token` properties as described here: https://docs.snowflake.com/en/developer-guide/jdbc/jdbc-parameters

I took a quick look at the code and it seems simple to add those two properties. I could submit a PR, but I'm not sure how the Trino library will handle not having a `connection-user` and `connection-password` since those are not needed.

**Comment by YechielLeader** [comment]
Also, how will it approve the Oauth, dont it need UI ?
